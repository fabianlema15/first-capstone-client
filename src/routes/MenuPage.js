import React from 'react';
import GenericContext from '../contexts/GenericContext'
import MenuContent from '../components/Menu/Menu'
import GenericApiService from '../services/generic-api-service';
import AuthApiService from '../services/auth-api-service';

class MenuPage extends React.Component{
  static defaultProps = {
    ROUTE : 'logins'
  }

  state = {
    userCode: null
  };

  setUserCode = userCode => {
    this.setState({ userCode })
  }


  setMenuList = objList => {
    this.setState({ objList })
  }

  setError = error => {
    this.setState({ error: error.error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  checkUserCode = user_code => {
    GenericApiService.getAll(`users/is/${user_code}`)
      .then(result => {
        this.setState({
          userCode: user_code
        })
      })
      .catch(this.setError)
  }

  checkMenu = password => {
    const loginData = {
      user_code: this.state.userCode,
      password: password,
    }
    AuthApiService.login(loginData)
      .then(result => {
        this.props.history.push(`/orders`)
      })
      .catch(this.setError)
  }

  render() {
    const value = {
      clearError: this.clearError,
      userCode: this.state.userCode,
      setUserCode: this.setUserCode,
      error: this.state.error,
      checkUserCode: this.checkUserCode,
      checkMenu: this.checkMenu
    }

    return (
      <GenericContext.Provider value={value}>
        <MenuContent />
      </GenericContext.Provider>
    )
  }
}

export default MenuPage;
