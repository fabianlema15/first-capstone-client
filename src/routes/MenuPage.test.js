import React from 'react';
import ReactDOM from 'react-dom';
import MenuPage from './MenuPage'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';

describe('<MenuPage />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
      <MenuPage>
      </MenuPage>
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders MenuPage as expected', () => {
	    const tree = renderer.create(<BrowserRouter>
        <MenuPage>
        </MenuPage>
      </BrowserRouter>).toJSON();
	    expect(tree).toMatchSnapshot();
	  });

})
