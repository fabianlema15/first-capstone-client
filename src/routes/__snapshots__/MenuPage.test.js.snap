// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuPage /> Renders MenuPage as expected 1`] = `
<div>
  <nav>
    <div
      className="name-centered"
    >
      <h1>
        Main Menu
      </h1>
    </div>
  </nav>
  <main
    className="content-with-nav"
  >
    <section>
      <div
        className="main-item"
      >
        <div
          className="menu-item"
        >
          Change Password
          <button
            className="border-gray"
            onClick={[Function]}
          >
            <img
              alt="Icon"
              src="menu_icon.png"
            />
          </button>
        </div>
      </div>
      <div
        className="main-item"
      >
        <div
          className="menu-item"
        >
          Logout
          <button
            className="border-gray"
            onClick={[Function]}
          >
            <img
              alt="Icon"
              src="menu_icon.png"
            />
          </button>
        </div>
      </div>
    </section>
  </main>
</div>
`;
